lib/scraper_test.rb

```diff
@@ -30,17 +31,41 @@ def install_tasks
           it 'should contain the expected data' do
             # TODO: Make this configurable
             Dir['test/data/*.yml'].each do |file|
-              yaml_data = YAML.load_file(file).to_h
-              url = yaml_data[:url]
-              class_to_test = Object.const_get(yaml_data[:class])
-              VCR.use_cassette(File.basename(url)) do
-                response = class_to_test.new(response: Scraped::Request.new(url: url).response)
-                response.to_h.must_equal yaml_data[:to_h]
+              VCR.use_cassette(File.basename(file)) do
+                test = Instructions.new(file)
+                response = test.class_to_test.new(response: Scraped::Request.new(url: test.url).response)
+                response.to_h.must_equal test.expected
               end
             end
           end
         end
       end
     end
   end
+
+  class Instructions
+    def initialize(file)
```

Comment:
This hides slightly what you're actually expecting to be passed here. If this is a filename, then it would be better to name it as that, so that someone reading the code can see clearly that that's what they should be passing, rather than have to puzzle it out.